### PR TITLE
Allow using a local view_component gem for test/local development

### PR DESCRIPTION
**Pull request by manuelpuyol**
Related to https://github.com/github/view_component/issues/525

If `VIEW_COMPONENT_PATH` env var is set, it will install the gem in that location instead of using RubyGems

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -16,6 +16,7 @@ gem "webpacker", "~> 5.0"
 # Reduces boot times through caching; required in config/boot.rb
 gem "bootsnap", ">= 1.4.2", require: false
 
+gem "view_component", path: ENV["VIEW_COMPONENT_PATH"] if ENV["VIEW_COMPONENT_PATH"]
```
Comment (BlakeWilliams) [inline Gemfile:19]:
```suggestion
gem "view_component", github: "github/view_component", branch: "main"
```

Would this work while also allowing https://bundler.io/man/bundle-config.1.html#LOCAL-GIT-REPOS to work?

Comment (srt32) [reply Gemfile:19]: were we getting view_components for free from Rails before this change?

Comment (BlakeWilliams) [reply Gemfile:19]: I think it may have been coming from the `gemspec` before this addition.

Comment (manuelpuyol) [reply Gemfile:19]: I'm not sure that we want to **always** override the `view_component` defined in `.gemspec`
I don't know what the implications would be

**Comment by manuelpuyol** [comment]
@BlakeWilliams may I have a 👍  here?

**Review by BlakeWilliams — APPROVED** [review]

**Pull request merged**